broadcast/src/linked/signer/actor.rs

```diff
@@ -0,0 +1,1010 @@
+                mail = self.mailbox_receiver.next() => {
+                    let msg = match mail {
+                        Some(msg) => msg,
+                        None => {
+                            error!("Mailbox receiver failed");
+                            break;
+                        }
+                    };
+                    match msg {
+                        Message::Broadcast{ payload_digest, result } => {
+                            debug!("Mailbox: Broadcast");
+                            if self.coordinator.is_sequencer(self.epoch, &self.crypto.public_key()).is_none() {
+                                error!("Not a sequencer in the current epoch");
+                                continue;
+                            }
+
+                            // Initialize my journal if it does not exist
+                            self.journal_prepare(&self.crypto.public_key()).await;
+
+                            // Broadcast the message
+                            if let Err(e) = self.broadcast_new(payload_digest, result, &mut link_sender).await {
+                                error!("Failed to broadcast new: {:?}", e);
+                                continue;
+                            }
+                        }
+                        Message::Verified{ context, payload_digest } => {
+                            debug!("Mailbox: Verified");
+                            if let Err(e) = self.handle_app_verified(&context, &payload_digest, &mut ack_sender).await {
+                                error!("Failed to handle app-verified: {:?}", e);
+                                continue;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    ////////////////////////////////////////
+    // Handling
+    ////////////////////////////////////////
+
+    /// Handles a verified message from the application.
+    ///
+    /// This is called when the application has verified a payload.
+    /// The chunk will be signed if it matches the current tip.
+    async fn handle_app_verified(
+        &mut self,
+        context: &Context,
+        payload_digest: &D,
+        ack_sender: &mut impl Sender,
+    ) -> Result<(), Error> {
+        // Get the tip
+        let Some(chunk) = self.tip_man.get_chunk(&context.sequencer) else {
+            return Err(Error::AppVerifiedNoTip);
+        };
+
+        // Return early if the height does not match
+        if chunk.height != context.height {
+            return Err(Error::AppVerifiedHeightMismatch);
+        }
+
+        // Return early if the payload digest does not match
+        let chunk_digest = D::read_from(&mut chunk.payload_digest.clone()).unwrap();
+        if chunk_digest != *payload_digest {
+            return Err(Error::AppVerifiedPayloadMismatch);
+        }
+
+        // Construct partial signature
+        let Some(share) = self.coordinator.share(self.epoch) else {
```

Comment:
I recommend adding some documentation to explain why your current storage approach won't lead to double-signing. AFAICT it won't but think explaining why is worth it.